src/page/slot.rs

```diff
@@ -76,14 +76,26 @@ impl<C: cfg::Config> Generation<C> {
     }
 }
 
-impl<T, C: cfg::Config> Slot<T, C> {
-    pub(in crate::page) fn new(next: usize) -> Self {
-        Self {
-            lifecycle: AtomicUsize::new(0),
-            item: CausalCell::new(None),
-            next: CausalCell::new(next),
-            _cfg: PhantomData,
-        }
+// Slot methods which should work across all trait bounds
+impl<T, C> Slot<T, C>
+where
+    C: cfg::Config,
+{
+    #[inline(always)]
+    pub(super) fn next(&self) -> usize {
+        self.next.with(|next| unsafe { *next })
+    }
+
+    #[inline(always)]
+    pub(super) fn value(&self) -> Option<&T> {
+        self.item.with(|item| unsafe { Some(&*item) })
```

Comment:
i don't know if it makes sense for this to still return `Option` --- it will now always return `Some`, and when the value type is `Option` (e.g. in the `Slab` case), this will return `Some(&None)` if the slot is not in use. i think this should just return `&T` instead --- `Slab` callers can then use `Option::as_ref` to turn the `&Option<T>` into an `Option<&T>`.
```suggestion
    pub(super) fn value(&self) -> &T {
        self.item.with(|item| unsafe { &*item })
```